geomesa-kafka-bundle/geomesa-kafka-processors/src/main/scala/org/geomesa/nifi/processors/kafka/GetGeoMesaKafkaRecord.scala

```diff
@@ -139,95 +134,111 @@ class GetGeoMesaKafkaRecord extends AbstractProcessor {
       val opts = SimpleFeatureConverterOptions(encoding = encoding, visField = vis, userDataField = userData)
       converter = SimpleFeatureRecordConverter(sft, opts)
       schema = factory.getSchema(Collections.emptyMap[String, String], converter.schema)
-      fs = ds.getFeatureSource(typeName)
-      fs.addFeatureListener(listener)
-    } catch {
-      case NonFatal(e) => CloseWithLogging(ds); ds = null; throw e
+      consumer = ds.createConsumer(typeName, groupId, processor)
```

Comment:
Does a consumer require any clean up when one is done with it?

Reply:
yes, it gets closed down on line 216 when the processor is stopped